packages/ai/src/ai/common/chat.py

```
@@ -451,6 +478,86 @@ def chat(self, question: Question) -> Answer:
+        prompt_tokens = self.getTokens(prompt)
+        if prompt_tokens >= self._modelTotalTokens - 100:
+            debug(f'Warning: Prompt ({prompt_tokens} tokens) exceeds input allocation ({self._modelTotalTokens} tokens)')
+
+        response = self._chat_stream_with_retries(prompt, on_chunk)
+
+        try:
+            result_tokens = self.getTokens(response)
+            if prompt_tokens + result_tokens >= self._modelTotalTokens - 5:
+                debug(f'Warning: Result ({result_tokens} tokens) was probably truncated')
+        except Exception:
+            pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)